utilities/post_migration.py

```
@@ -437,18 +442,18 @@ def get_matching_interface_with_ip():
                         LOGGER.warning(f"Interface {interface_name} not found yet")
                         return None  # Retry
 
-                    # Check if expected IP is found
+                    # Check if expected IP is found and extract its subnet mask
                     interface_ips = matching_interface.get("ip_addresses", [])
-                    all_ips = [ip_info.get("ip_address") for ip_info in interface_ips]
 
-                    found_expected_ip = any(ip_info.get("ip_address") == expected_ip for ip_info in interface_ips)
+                    for ip_info in interface_ips:
+                        if ip_info.get("ip_address") == expected_ip:
```

Ruff (0.15.9)

[warning] Function definition does not bind loop variable `expected_ip`

(B023)